include/drivers/irq/hifive.h

```diff
@@ -36,12 +36,19 @@
 #define PLIC_THRES_PER_CONTEXT  0x1000
 #define PLIC_THRES_CLAIM        0x4
 
-#ifdef CONFIG_PLAT_HIFIVE
+#if defined(CONFIG_PLAT_HIFIVE) || defined(CONFIG_PLAT_POLARFIRE)
+
+#if defined(CONFIG_PLAT_HIFIVE)
+#define PLIC_NUM_INTERRUPTS 53
+#elif defined(CONFIG_PLAT_POLARFIRE)
+#define PLIC_NUM_INTERRUPTS 93
+#endif
+
 /* SiFive U54-MC has 5 cores, and the first core does not
  * have supervisor mode. Therefore, we need to compensate
  * for the addresses.
  */
-#define PLIC_NUM_INTERRUPTS 53
```

Comment:
It should be possible to replace this block with:
```
#define PLIC_NUM_INTERRUPTS CONFIGURE_PLIC_MAX_NUM_INT
```

Reply:
So the `CONFIGURE_PLIC_MAX_NUM_INT` is set by the `declare_default_headers` function that is called in the platforms `config.cmake` file with the `PLIC_MAX_NUM_INT` argument. These get defined in the `platform_gen.h` file. If I do this, and include `plat/default/plat/machine.h` (which includes `plat/machine/devices_gen.h` and `plat/platform_gen.h`) in the plic driver header, I get a circular dependency error.

Reply:
Ok, maybe we can fix the order some other time then.  Thanks for trying though!

Reply:
What about PLIC_MAX_IRQ? It looks like platform_gen.h declares it before importing drivers/irq/hifive.h? Don't worry if it doesn't work though.
```
enum IRQConstants {
    PLIC_IRQ_OFFSET = 0,
    PLIC_MAX_IRQ = PLIC_IRQ_OFFSET + (@CONFIGURE_PLIC_MAX_NUM_INT@),
#ifdef ENABLE_SMP_SUPPORT
    INTERRUPT_IPI_0,
    INTERRUPT_IPI_1,
#endif
    INTERRUPT_CORE_TIMER,
    maxIRQ = INTERRUPT_CORE_TIMER,
} platform_interrupt_t;

enum irqNumbers {
    irqInvalid = 0
};

#define KERNEL_TIMER_IRQ INTERRUPT_CORE_TIMER
#define IRQ_CNODE_SLOT_BITS (@CONFIGURE_IRQ_SLOT_BITS@)

#include <@CONFIGURE_INTERRUPT_CONTROLLER@>
```

Reply:
You shouldn't need to include anything extra

Reply:
That seems to work. Do you think it'd be best to just have 
```
#define PLIC_NUM_INTERRUPTS PLIC_MAX_IRQ
```
and drop the default 511? Each platform should set the max IRQ right?

Reply:
@yyshen how come you added a default 511 definition here?